src/Microsoft.Extensions.Logging.Testing/AssemblyTestLog.cs

```diff
@@ -51,60 +58,85 @@ public IDisposable StartTestLog(ITestOutputHelper output, string className, out
                 stopwatch.Stop();
                 _globalLogger.LogInformation("Finished test {testName} in {duration}s", testName, stopwatch.Elapsed.TotalSeconds);
                 logger.LogInformation("Finished test {testName} in {duration}s", testName, stopwatch.Elapsed.TotalSeconds);
+                scope.Dispose();
                 factory.Dispose();
+                (serviceProvider as IDisposable)?.Dispose();
             });
         }
 
         public ILoggerFactory CreateLoggerFactory(ITestOutputHelper output, string className, [CallerMemberName] string testName = null) =>
-            CreateLoggerFactory(output, className, LogLevel.Debug, testName);
+            CreateLoggerFactory(output, className, LogLevel.Trace, testName);
 
         public ILoggerFactory CreateLoggerFactory(ITestOutputHelper output, string className, LogLevel minLogLevel, [CallerMemberName] string testName = null)
         {
-            var serviceCollection = new ServiceCollection();
-            serviceCollection.AddLogging(builder =>
-            {
-                builder.SetMinimumLevel(minLogLevel);
-                if (output != null)
-                {
-                    builder.AddXunit(output, minLogLevel);
-                }
-            });
+            return CreateLoggerServices(output, className, minLogLevel, testName).GetRequiredService<ILoggerFactory>();
+        }
 
-            var loggerFactory = serviceCollection.BuildServiceProvider().GetRequiredService<ILoggerFactory>();
+        public IServiceProvider CreateLoggerServices(ITestOutputHelper output, string className, LogLevel minLogLevel, [CallerMemberName] string testName = null)
+        {
             // Try to shorten the class name using the assembly name
             if (className.StartsWith(_assemblyName + "."))
             {
                 className = className.Substring(_assemblyName.Length + 1);
             }
 
+            SerilogLoggerProvider serilogLoggerProvider = null;
             if (!string.IsNullOrEmpty(_baseDirectory))
             {
                 var testOutputFile = Path.Combine(_baseDirectory, _assemblyName, className, $"{testName}.log");
 
-                AddFileLogging(loggerFactory, testOutputFile);
+                serilogLoggerProvider = ConfigureFileLogging(testOutputFile);
             }
 
-            return loggerFactory;
-        }
-
-        public static AssemblyTestLog Create(string assemblyName, string baseDirectory)
-        {
             var serviceCollection = new ServiceCollection();
+            serviceCollection.AddLogging(builder =>
+            {
+                builder.SetMinimumLevel(minLogLevel);
 
-            // Let the global logger log to the console, it's just "Starting X..." "Finished X..."
-            serviceCollection.AddLogging(builder => builder.AddConsole());
```

Comment:
What do we need to do to enable xunit's long running test reporting now that this is being removed?

Reply:
Globally: https://github.com/aspnet/Coherence-Signed/issues/573

Locally: Just ensure the build drops an xunit.runner.json in the output directory with the [necessary config](https://xunit.github.io/docs/configuring-with-json.html). See https://github.com/aspnet/SignalR/blob/ce39b47e7de0f3baf219732801481497d2f1591e/test/xunit.runner.json for more detail